File: ios/oldNative/JSColorGradientType.m
//
//  JSColorGradientType.m
//  Supermap
//
//  Created by supermap on 2018/8/9.
//  Copyright © 2018年 Facebook. All rights reserved.
//

#import "JSColorGradientType.h"

@implementation JSColorGradientType
RCT_EXPORT_MODULE();
-(NSDictionary*)constantsToExport{
    return @{@"BLACKWHITE":@(0),
             @"REDWHITE":@(1),
             @"GREENWHITE":@(2),
             @"BLUEWHITE":@(3),
             @"YELLOWWHITE":@(4),
             @"PINKWHITE":@(5),
             @"CYANWHITE":@(6),
             @"REDBLACK":@(7),
             @"GREENBLACK":@(8),
             @"BLUEBLACK":@(9),
             @"YELLOWBLACK":@(10),
             @"PINKBLACK":@(11),
             @"CYANBLACK":@(12),
             @"YELLOWRED":@(13),
             @"YELLOWGREEN":@(14),
             @"YELLOWBLUE":@(15),
             @"GREENBLUE":@(16),
             @"GREENRED":@(17),
             @"BLUERED":@(18),
             @"PINKRED":@(19),
             @"PINKBLUE":@(20),
             @"CYANBLUE":@(21),
             @"CYANGREEN":@(22),
             @"RAINBOW":@(23),
             @"GREENORANGEVIOLET":@(24),
             @"TERRAIN":@(25),
             @"SPECTRUM":@(26)
             };
}
@end

File: android/src/main/java/com/supermap/rnsupermap/JSColorGradientType.java
package com.supermap.rnsupermap;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Myself on 2017/8/15.
 */

public class JSColorGradientType extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS="JSColorGradientType";

    public JSColorGradientType(ReactApplicationContext context){
        super(context);
    }

    @Override
    public String getName(){return REACT_CLASS;}

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("BLACKWHITE", 0);
        constants.put("REDWHITE", 1);
        constants.put("GREENWHITE", 2);
        constants.put("BLUEWHITE", 3);
        constants.put("YELLOWWHITE", 4);
        constants.put("PINKWHITE", 5);
        constants.put("CYANWHITE", 6);
        constants.put("REDBLACK", 7);
        constants.put("GREENBLACK", 8);
        constants.put("BLUEBLACK", 9);
        constants.put("YELLOWBLACK", 10);
        constants.put("PINKBLACK", 11);
        constants.put("CYANBLACK", 12);
        constants.put("YELLOWRED", 13);
        constants.put("YELLOWGREEN", 14);
        constants.put("YELLOWBLUE", 15);
        constants.put("GREENBLUE", 16);
        constants.put("GREENRED", 17);
        constants.put("BLUERED", 18);
        constants.put("PINKRED", 19);
        constants.put("PINKBLUE", 20);
        constants.put("CYANBLUE", 21);
        constants.put("CYANGREEN", 22);
        constants.put("RAINBOW", 23);
        constants.put("GREENORANGEVIOLET", 24);
        constants.put("TERRAIN", 25);
        constants.put("SPECTRUM", 26);
        return constants;
    }
}